### ISSUE TITLE
Research object detection

### ISSUE THREAD

**Issue by Siliconlad** [automat-ed/autonomous#4]
For Demo 2, we need to be able to show that the robot can detect objects (including humans) within the simulation. Before we can do this, we need to conduct some research into how to do this. In addition, we need to compile a list of required sensors to pass to the hardware team so that they can design the robot.

# Task
To conduct research into different methods for detecting objects within the simulation. All research should be documented in the [Google Drive folder](https://drive.google.com/drive/folders/1m9Mdte_oveNHa0qymPkkvpYUwnPCyi6p?usp=sharing). In addition, a list of required sensors should be compile along with their position for optimal use.

**Comment by Siliconlad** [comment]
The expert for computer vision has provided some useful advice on how to approach this problem.

Two weeks ago he talked about Facebook's detectron2 repo and also suggested looking at pre-existing datasets (e.g kitty or waymo) as well leaderboards for various challenges that may point us to useable software.
> Bounding box should be sufficient. There are a lot of prebuilt neural networks that do this stuff but they also do other relevant things, like detecting what is street and what is a building, other objects etc. Facebook has bundled a few of these networks in their detectron2 repo (https://github.com/facebookresearch/detectron2) but I think if you do some research you might find projects/networks that suit your project even better. Keywords are semantic segmentation, object detection, automotive. Plus you can have a look at the kitty, waymo, etc datasets. They sometimes have challenges with leaderboards which in turn link to usable software.

This week he's suggested looking into YOLO as an alternative to detectron. He also suggested using a process of elimination to decide whether our robot is safe to move around a detected object (via LiDAR) by making sure the object is not moving and not recognized as human. He also mentioned [paperswithcode](https://paperswithcode.com/) as a good place to look for existing implementations of algorithms.
> there is ROS perception but it is mostly consisting out of bridges and specifications. I would use it but it does not shield you from the complexity. You still need to integrate the algo/software you want to use. Last time I spoke about detectron, but you could also try YOLO (https://www.youtube.com/watch?v=MPU2HistivI). It can detect dogs and humans which means you could also conclude by elimination. I think there is v5 out there. Here an article comparing yolo with faster rcnn, which is what detectron is based on: https://towardsdatascience.com/yolov5-compared-to-faster-rcnn-who-wins-a771cd6c9fb4 (Say you detect an obstacle with the lidar and you know its not a human or a dog and its not moving, you can savely move around it)

Please also see the [Drive folder for meeting minutes and questions](https://docs.google.com/document/d/1XrZ1vRc6M1mgdpx8cbclVnf0KHoxaUY7dmiIMC2b2r0/edit?usp=sharing) asked.

**Comment by Siliconlad** [comment]
For this issue we should prioritize detecting humans to ensure a certain a certain level of safety while operating the robot.

**Comment by Siliconlad** [comment]
This has been postponed to Demo 3.

**Comment by Siliconlad** [comment]
Postponed until Demo 4.

**Comment by Siliconlad** [comment]
Closing this as the relevant task is being tracked by #17.
